scripts/signature.py

```diff
@@ -85,6 +85,15 @@ def diff(args):
     else:
         print(f"{bcolors.BOLD}No differences found.")
 
+    if exit_status:
+        print(
+            f""" 
+            {bcolors.BOLD} Non compatible changes in function signatures have been detected.
```

Comment:
Not sure I follow. Are you saying that this compares the PR state with fresh main, not with the point where the PR was forked? Should we compare with that?

Reply:
Yes, it used to use main - I have changed it to use fork-point, nevertheless It might still be helpful to print these messages even with fork-point.

Reply:
Thanks for the change. Still trying to understand this. If we're now comparing with the commit in which you're based on, in which cases would it trigger this failure? Wouldn't this only fail now in case there's an actual signature comparison failure?

Reply:
Yes it will only fail if your changes modify the signatures of any function inadvertently or otherwise. Without this change, its possible for the job to fail if a signature was modified in the main branch and you hadnt rebased your branch with those changes.

Reply:
Yes. My point is that now with this change, we don't need this error message anymore (unless I'm missing something?)

Reply:
Sure I will remove this then; I added it just as an additional fyi.

Reply:
Thanks!

nit: maybe "incompatible" instead?